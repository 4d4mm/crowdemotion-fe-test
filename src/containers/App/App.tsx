import * as React from "react";
import { Route, BrowserRouter } from "react-router-dom";
import PrivateRoute from "../../containers/PrivateRoute/PrivateRoute";
import LoginPage from "../../pages/LoginPage/LoginPage";
import SurveyForm from "../../pages/NewSurveyForm/NewSurveyForm";
import SurveyDetails from "../../pages/SurveyDetails/SurveyDetails";
import SurveyList from "../../pages/ListSurveys/ListSurveys";

type AppProps = {
    router?: React.ComponentType
}

const App:React.SFC<AppProps> = ({ router:Router = BrowserRouter }) => (
    <div className="App">
      <Router>
        <Route exact path="/login" component={LoginPage} />
        <PrivateRoute exact path="/" component={SurveyList} />
        <PrivateRoute exact path="/new-survey" component={SurveyForm} />
        <PrivateRoute exact path="/surveys/:surveyId" component={SurveyDetails} />
      </Router>
    </div>
);

export default App;