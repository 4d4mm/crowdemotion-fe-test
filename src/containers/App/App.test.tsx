import React from "react";
import { act } from 'react-dom/test-utils';
import { MemoryRouter } from "react-router-dom";
import { mount } from "enzyme";
import App from "./App";

const ConfiguredMenoryRouter = ({ children }) =>(<MemoryRouter keyLength={0} initialEntries={[{pathname: '/'}]}>{children}</MemoryRouter>)

describe("App", () => {
  it("renders with default ui", () => {
    act(()=>{
        const renderedApp = mount(<App router={ConfiguredMenoryRouter}/>);
        expect(renderedApp).toMatchSnapshot();
    })
  });
});
