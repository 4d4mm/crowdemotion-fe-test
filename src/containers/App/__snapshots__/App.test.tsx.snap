// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders with default ui 1`] = `
<App
  router={[Function]}
>
  <div
    className="App"
  >
    <ConfiguredMenoryRouter>
      <MemoryRouter
        initialEntries={
          Array [
            Object {
              "pathname": "/",
            },
          ]
        }
        keyLength={0}
      >
        <Router
          history={
            Object {
              "action": "REPLACE",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "pathname": "/login",
                  "search": "",
                  "state": undefined,
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/login",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <Route
            component={[Function]}
            exact={true}
            path="/login"
          />
          <PrivateRoute
            component={[Function]}
            exact={true}
            path="/"
          >
            <Route
              exact={true}
              path="/"
              render={[Function]}
            >
              <Unauthenticated
                history={
                  Object {
                    "action": "REPLACE",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "pathname": "/login",
                        "search": "",
                        "state": undefined,
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/login",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <Redirect
                  to="/login"
                >
                  <Lifecycle
                    onMount={[Function]}
                    onUpdate={[Function]}
                    to="/login"
                  />
                </Redirect>
              </Unauthenticated>
            </Route>
          </PrivateRoute>
          <PrivateRoute
            component={[Function]}
            exact={true}
            path="/new-survey"
          >
            <Route
              exact={true}
              path="/new-survey"
              render={[Function]}
            />
          </PrivateRoute>
          <PrivateRoute
            component={[Function]}
            exact={true}
            path="/surveys/:surveyId"
          >
            <Route
              exact={true}
              path="/surveys/:surveyId"
              render={[Function]}
            />
          </PrivateRoute>
        </Router>
      </MemoryRouter>
    </ConfiguredMenoryRouter>
  </div>
</App>
`;
